amd64/root/firmware-update.sh

```diff
@@ -156,11 +160,13 @@ if [[ ! -f ${FLASHER_PARAM_VALUES[-f]} ]]; then
     echo " "
     echo "Downloading..."
     echo " "
-    curl --fail --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_BASE%/}/$fileName" && [[ -f ${FLASHER_PARAM_VALUES[-f]} ]]
-    delete_and_exit_on_error "${FLASHER_PARAM_VALUES[-f]}" "Download Error! Please re-run this script..."
+    curl --fail --silent --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_STABLE_BASE%/}/$fileName" || \
+    curl --fail --silent --output "${FLASHER_PARAM_VALUES[-f]}" "${FW_ONLINE_BETA_BASE%/}/$fileName"
+    [[ -f ${FLASHER_PARAM_VALUES[-f]} ]] || delete_and_exit_on_error "${FLASHER_PARAM_VALUES[-f]}" "Download Error! Please re-run this script..."
```

Comment:
Here you are breaking the error handling.
The old version did the following:
- try the download and check for file existence
- if one of them failed the script will delete the file (if it exists) and end with an error

The `delete_and_exit_on_error` is using the returncode of the last command to decide what to do. By breaking the download and filecheck into two lines the behaviour doesn't consider the results of curl anymore ...